liquibase-core/src/main/java/liquibase/command/core/TagExistsCommandStep.java

```diff
@@ -1,52 +1,47 @@
 import liquibase.command.*;
-import liquibase.configuration.ConfigurationValueObfuscator;
-import liquibase.exception.CommandExecutionException;
+import liquibase.database.Database;
+import liquibase.lockservice.LockService;
+import liquibase.lockservice.LockServiceFactory;
 
-public class TagExistsCommandStep extends AbstractCliWrapperCommandStep {
+import java.util.Arrays;
+import java.util.List;
 
-    public static final String[] COMMAND_NAME = {"tagExists"};
+public class TagExistsCommandStep  extends AbstractCommandStep {
+
+    protected static final String[] COMMAND_NAME = {"tagExists"};
 
-    public static final CommandArgumentDefinition<String> URL_ARG;
-    public static final CommandArgumentDefinition<String> DEFAULT_SCHEMA_NAME_ARG;
-    public static final CommandArgumentDefinition<String> DEFAULT_CATALOG_NAME_ARG;
-    public static final CommandArgumentDefinition<String> USERNAME_ARG;
-    public static final CommandArgumentDefinition<String> PASSWORD_ARG;
     public static final CommandArgumentDefinition<String> TAG_ARG;
-    public static final CommandArgumentDefinition<String> DRIVER_ARG;
-    public static final CommandArgumentDefinition<String> DRIVER_PROPERTIES_FILE_ARG;
+    public static final CommandResultDefinition<Boolean> TAG_EXISTS_RESULT;
 
     static {
         CommandBuilder builder = new CommandBuilder(COMMAND_NAME);
-        URL_ARG = builder.argument(CommonArgumentNames.URL, String.class).required()
-                .description("The JDBC database connection URL").build();
-        DEFAULT_SCHEMA_NAME_ARG = builder.argument("defaultSchemaName", String.class)
-                .description("The default schema name to use for the database connection").build();
-        DEFAULT_CATALOG_NAME_ARG = builder.argument("defaultCatalogName", String.class)
-                .description("The default catalog name to use for the database connection").build();
-        DRIVER_ARG = builder.argument("driver", String.class)
-                .description("The JDBC driver class").build();
-        DRIVER_PROPERTIES_FILE_ARG = builder.argument("driverPropertiesFile", String.class)
-                .description("The JDBC driver properties file").build();
-        USERNAME_ARG = builder.argument(CommonArgumentNames.USERNAME, String.class)
-                .description("Username to use to connect to the database").build();
-        PASSWORD_ARG = builder.argument(CommonArgumentNames.PASSWORD, String.class)
-                .description("Password to use to connect to the database")
-                .setValueObfuscator(ConfigurationValueObfuscator.STANDARD)
-                .build();
-        TAG_ARG = builder.argument("tag", String.class).required()
-                .description("Tag to check").build();
+        TAG_ARG = builder.argument("tag", String.class).required().description("Tag to check").build();
+
+        TAG_EXISTS_RESULT = builder.result("tagExistsResult", Boolean.class).build();
```

Comment:
What is this new property? Can you add a description to it or make it hidden?

Reply:
This new property is the output from execution that can obtained by calling `commandResults.getResult(TagExistsCommandStep.TAG_EXISTS_RESULT); `
I'm adding the description .

Reply:
I am a bit confused though. Why does the tag exists command define its own result as a command argument?

Reply:
It's not an argument.. It's a result :)
*Argument: commandScope.getArgumentValue(TagCommandStep.TAG_ARG)));
* Result:  resultsBuilder.addResult(TAG_EXISTS_RESULT, true);

And at Liquibase class you can see both of them in use:
```
        CommandResults commandResults = new CommandScope("tagExists")
                .addArgumentValue(DbUrlConnectionCommandStep.DATABASE_ARG, database)
                .addArgumentValue(**TagExistsCommandStep.TAG_ARG**, tagString)
                .execute();
        return commandResults.getResult(**TagExistsCommandStep.TAG_EXISTS_RESULT**);
```

Reply:
Got it, that makes more sense. Thanks for adding the description!